### PR TITLE
upgrade all deps to latest semver, initial working tests

### PR DESCRIPTION
**Pull request by amilajack**
## Tasks
- [x] upgrade infra (eslint, babel, flow)
- [x] refactor compat related code to make it more testable
- [x] initial passing tests
- [x] merge with master
- [x] migrate all chai assertions to jest assertions
- [x] remove hacky top-level code and code that depends on it

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,19 +1,10 @@
...
-    ["flow-runtime", {
-      "assert": true,
-      "annotate": true
-    }],
-    */
-    "transform-decorators-legacy",
-    "transform-runtime",
-    "add-module-exports",
-    "transform-flow-strip-types"
+    "@babel/plugin-syntax-flow",
```
Comment (vikr01) [inline .babelrc]: Can remove this.

Comment (amilajack) [reply .babelrc]: This as in?

Comment (vikr01) [reply .babelrc]: `@babel/plugin-syntax-flow`

Comment (amilajack) [reply .babelrc]: Oh my bad. skimmed and read that as preset flow

**Pull request merged**